pkgs/stdenv/adapters.nix

```diff
@@ -2,7 +2,31 @@
    a new stdenv with different behaviour, e.g. using a different C
    compiler. */
 
-pkgs:
+{ lib, pkgs, config }:
+
+let
+  # N.B. Keep in sync with default arg for stdenv/generic.
+  defaultMkDerivationFromStdenv = import ./generic/make-derivation.nix { inherit lib config; };
+
+  # Low level function to help with overriding `mkDerivationFromStdenv`. One
+  # gives it the old stdenv arguments and a "continuation" funciont, and
+  # underneath the final stdenv argument it yields to the continuation to do
+  # whatever it wants with old `mkDerivation` (old `mkDerivationFromStdenv`
+  # applied to the *new, final* stdenv) provided for convience.
+  withOldMkDerivation = stdenvSuperArgs: k: stdenvSelf: let
```

Comment:
First argument name is confusing me a little.
Could it just be `stdenvSuper`?

Reply:
It's not the stdenv itself, but the arguments passed to `generic/default.nix`.

Reply:
Ah yes!